### ISSUE TITLE
`org-roam-node-doom-filetitle` makes calls to `org-roam-node-find` slow in Doom

### ISSUE THREAD

**Issue by harabat** [org-roam/org-roam#2186]
### Description

Calling `org-roam-node-find` should be fast, especially since #1963 (related to issue #1565). However, this is very slow (>5 seconds) in Doom. 

Running the profiler, I can pinpoint this to `org-roam-node-doom-filetitle` defined in [contrib-roam2.el](https://github.com/doomemacs/doomemacs/blob/3b789752a30419535b328c605957b6b43a9d47fa/modules/lang/org/autoload/contrib-roam2.el#L6-L12) by @Wetlize in 07/2021. Let me know if I should open the issue at [doomemacs/doomemacs](https://github.com/doomemacs/doomemacs) instead. 

Having ruled out my config's role in this after bisecting, I tried to simplify the method in `~/.emacs.d/modules/lang/org/autoload/contrib-roam2.el` to:

```elisp
(cl-defmethod org-roam-node-doom-filetitle ((node org-roam-node))
  (or (org-roam-get-keyword "TITLE" (org-roam-node-file node))
      "")))
```

However, the problem persists. 

I wasn't able to implement #1963's solution by myself, sorry. 

#### Steps to Reproduce

1. Load Doom Emacs
2. Run `org-roam-node-find`

<details>
<summary>Environment</summary>

- Emacs: GNU Emacs 28.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.33, cairo version 1.17.6)
 of 2022-04-28
- Framework: Doom
- Org: Org mode version 9.6 (9.6-??-971eb68 @ /home/warehouse/.emacs.d/.local/straight/build-28.1/org/)
- Org-roam: 3615259
</details>

<details>
<summary>Profiler</summary>

                                                                399,063,340  99% - command-execute
                                                                399,063,340  99%  - funcall-interactively
                                                                395,364,869  99%   - org-roam-node-find
                                                                395,364,869  99%    - org-roam-node-read
                                                                394,234,020  98%     - org-roam-node-read--completions
                                                                369,937,050  92%      - #<compiled -0xa8c40c191df195e>
                                                                369,829,338  92%       - org-roam-node-read--to-candidate
                                                                369,829,338  92%        - apply
                                                                369,475,578  92%         - +org--roam-fix-completion-width-for-vertico-a
                                                                369,364,698  92%          - let
                                                                369,148,218  92%           - let*
                                                                369,028,890  92%            - unwind-protect
                                                                369,028,890  92%             - progn
                                                                369,028,890  92%              - apply
                                                                359,232,426  90%               - #<subr org-roam-node-read--to-candidate>
                                                                358,889,226  89%                - org-roam-node--format-entry
                                                                358,889,226  89%                 - funcall
                                                                350,236,842  87%                  - #<subr org-roam-node--format-entry>
                                                                350,112,260  87%                   - org-roam-format-template
                                                                334,997,604  83%                    - replace-regexp-in-string
                                                                325,467,609  81%                     - #<compiled 0x1bd1f116308f8218>
                                                                312,023,386  78%                      - #<compiled 0x64a549041498f66>
                                                                216,201,975  54%                       - org-roam-node-doom-hierarchy
                                                                216,201,975  54%                        - apply
                                                                216,084,759  54%                         - #<lambda -0x10a57ffcc1d7c51>
                                                                216,084,759  54%                          - progn
                                                                215,463,831  53%                           - let
                                                                203,601,799  51%                            - org-roam-node-doom-filetitle
                                                                203,601,799  51%                             - apply
                                                                203,488,807  50%                              - #<lambda -0x1f97d246a0d35ecd>
                                                                203,488,807  50%                               - progn
                                                                203,488,807  50%                                - or
                                                                203,488,807  50%                                 - if
                                                                 87,944,717  22%                                  - org-roam-get-keyword
                                                                 72,587,374  18%                                   - set-auto-coding
                                                                 72,587,374  18%                                      find-auto-coding
                                                                  3,029,787   0%                                   + #<compiled -0x1c8b3d5042a83097>
                                                                 11,310,128   2%                            - cond
                                                                  5,914,496   1%                             + concat
                                                                 35,473,167   8%                       - org-roam-node-doom-type
                                                                 35,473,167   8%                        - apply
                                                                 35,366,511   8%                         - #<lambda 0x1d28dd9a7e243e5d>
                                                                 35,366,511   8%                          - progn
                                                                 35,254,575   8%                           - let*
                                                                 35,039,143   8%                            - and
                                                                 34,787,431   8%                             - file-name-directory
                                                                     14,011   0%                              - file-relative-name
                                                                     14,011   0%                                 #<compiled 0xae96b2f6b4263>
                                                                    215,432   0%                              if
                                                                 27,272,308   6%                       - truncate-string-to-width
                                                                      2,720   0%                          char-width
                                                                  5,720,440   1%                         mapconcat
                                                                    921,888   0%                       + org-roam-node-doom-tags
                                                                  1,220,048   0%                        apply
                                                                      2,118   0%                        #<compiled 0x8c0ab2b8fa6a>
                                                                      2,091   0%                        match-data
                                                                      2,083   0%                       #<compiled 0xae96b2f4f9663>
                                                                  8,652,384   2%                  + if
                                                                 21,865,916   5%      - org-roam-node-list
                                                                 21,865,916   5%       - org-roam-db-query
                                                                 21,865,916   5%        - emacsql
                                                                 21,865,916   5%         - apply
                                                                 21,865,916   5%          - #<compiled -0x98ddd6fe938fbe4>
                                                                 12,358,640   3%           - emacsql-wait
                                                                 12,358,640   3%            - apply
                                                                 12,358,640   3%               #<compiled -0xbec5267d16c82b4>
                                                                  9,427,274   2%           - emacsql-parse
                                                                  9,427,274   2%            - apply
                                                                  9,427,274   2%               #<compiled 0x1056c041553a0cd9>
                                                                     79,959   0%           + apply
                                                                         43   0%           + emacsql-send-message
                                                                  2,176,312   0%      + seq-sort
                                                                      6,582   0%      + org-roam-node--process-display-format
                                                                  1,091,993   0%     + completing-read-default
                                                                  3,698,471   0%   + doom/toggle-profiler
                                                                     10,488   0% + redisplay_internal (C function)
                                                                      8,184   0% + evil-repeat-pre-hook
                                                                      4,224   0% + emojify-update-visible-emojis-background-after-command
                                                                      1,056   0%   winner-save-old-configurations
                                                                      1,024   0% + evil-escape-pre-command-hook
                                                                        184   0% + gcmh-register-idle-gc
                                                                         24   0% + eldoc-schedule-timer
                                                                         21   0% + #<compiled -0x87d9772881b1999>
                                                                          0   0%   ...
</details>

**Comment by jethrokuan** [comment]
Please file this ticket in Doom instead, as this is Doom specific.

**Comment by harabat** [comment]
The issue was resolved in [#6382](https://github.com/doomemacs/doomemacs/issues/6382).
